### PR TITLE
Hw05 avatars

### PR DESCRIPTION
**Pull request by OIgaB**

### CODE REVIEW

File: controllers/auth.js [hunk 1]
```diff
@@ -71,19 +78,71 @@ const logout = async(req, res) => {
...
+
+const avatarPath = path.resolve("public", "avatars"); // шлях до папки з файлом
+
+const updateAvatar = async(req, res) => {
+    const { _id } = req.user; // new ObjectId("64c6f0e733523b6f5a4ba4b8"),
+    const { path: oldPath, filename } = req.file; //  path до temp; filename - нова назва файлу
+    // req.file: {
+    //     fieldname: 'avatar',
+    //     originalname: 'drink.jpg',
+    //     encoding: '7bit',
```
Comment (ceREAL133) [inline controllers/auth.js]: delete comments

File: controllers/auth.js [hunk 2]
```diff
@@ -71,19 +78,71 @@ const logout = async(req, res) => {
...
+    await fs.rename(oldPath, newPath); // переміщення файла до public\avatars  (переміщення вже було реалізоване в Jimp), + видалення з temp оригінального
+
+    const avatarURL = path.join('avatars', filename); // записуємо шлях в body // папку 'public' не пишемо, вона вже вказана в мідлварі в app.js
+    // avatarURL: avatars\1691116849916-469261661_drink.jpg
+
+    await User.findByIdAndUpdate(_id, {avatarURL}); // знаючи id користувача, можемо перезаписати avatarURL
+    // {
+    //     _id: new ObjectId("64cc5caaf34040bef1e8dff5"),
+    //     email: 'example2@example.com',
+    //     password: '$2b$10$FgrVO/FkkpnVdzxCDP5WXObM8jp7sOmBwSs1nirQN5pr9g3Posd6a',
```
Comment (ceREAL133) [inline controllers/auth.js]: Delete

File: controllers/auth.test.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+// у відповіді повинен повертатися об'єкт user з 2 полями email и subscription з типом даних String
+
+    test("test login with correct data", async()=> {
+        const loginData = {
+            "email": "example2@example.com",
+            "password": "example2password"
+        };
+        const { statusCode, body } = await request(app).post("/api/auth/login").send(loginData);
+        // body: {
+        //     token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6IjY0Y2Y5Y2RiM2RkMDNkMjhhZjc4M2E2YyIsImlhdCI6MTY5MTMyODE3NSwiZXhwIjoxNjkxNDEwOTc1fQ.ivDVrkh3CRk2Ohb4CNMYXUInkj5dPJ6yf3nLzEg6IRo',
```
Comment (ceREAL133) [inline controllers/auth.test.js]: delete

File: controllers/auth.test.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        expect(body.token).toBeDefined();  //toBeDefined() checks if the value is not undefined.
+
+        expect(body.user.email).toBe(loginData.email);
+
+        const user = await User.findOne({email: loginData.email}); // перевіряємо чи записалось в БД
+        // user: {
+        //     _id: new ObjectId("64cf9cdb3dd03d28af783a6c"),
+        //     email: 'example2@example.com',
+        //     password: '$2b$10...',
+        //     subscription: 'starter',
```
Comment (ceREAL133) [inline controllers/auth.test.js]: delete

**Review by ceREAL133 — CHANGES_REQUESTED** [review]
Delete all useless comments

File: controllers/auth.js (hunk 1 above)
Comment (OIgaB) [reply controllers/auth.js]: Yes, sir!

File: controllers/auth.js (hunk 2 above)
Comment (OIgaB) [reply controllers/auth.js]: Yes, sir!

File: controllers/auth.test.js (hunk 3 above)
Comment (OIgaB) [reply controllers/auth.test.js]: Yes, sir!

File: controllers/auth.test.js (hunk 4 above)
Comment (OIgaB) [reply controllers/auth.test.js]: Yes, sir!

**Review by ceREAL133 — APPROVED** [review]

**Pull request merged**